app/controllers/application_controller.rb

```diff
@@ -20,28 +17,44 @@ def jsonapi_render(collection, filters)
   end
 
   def set_user
-    @current_user = if auth_token.present?
-        User.from_jwt(Authenticator.from_token(auth_token))
-      else
-        nil
-      end
+    return nil unless auth_token.present?
+
+    @current_user = User.where(
+                      id: panoptes.client.authenticated_user_id,
```

Comment:
no need to change this at all, i would have added a delegate `:authenicated_user_id` to the `PanoptesApi` class that forwarded to the client. this would then make this code, `panoptes.authenticated_user_id`.

https://apidock.com/rails/Module/delegate

I'd would have to inspect the `PanoptesApi` class to understand how this worked but the abstraction is neater imo.

Reply:
Implemented this, great idea.